atlas_core/serializers.py

```diff
@@ -0,0 +1,117 @@
+from .interfaces import ISerializerStrategy
+
+from flask import jsonify, current_app, request
+
+
+def simplify_obj(obj):
+    if hasattr(obj, "_asdict"):
+        """SQLAlchemy result sets"""
+        return obj._asdict()
+    else:
+        return repr(obj)
+
+
+def ensure_simple(obj, simplify_func=simplify_obj):
+    """Take an object and try to traverse it and ensuring all its
+    sub-objects are simple to serialize primitive objects. If not, call repr()
+    or some other function on them to convert them into a primitive. This is
+    slow and is best-effort: It's for stuff like error messages, showing the
+    user some unpredictable payload that may or may not contain random objects
+    of any type. It isn't really meant to be used on large and predictable
+    objects like API responses - in those cases you probably want to process
+    those manually and systematically to make sure no non-serializable items
+    are in there in the first place."""
+    if type(obj) in (str, int, float, bool):
+        return obj
+    elif obj is None:
+        return None
+    elif type(obj) == list:
```

Comment:
Does it make more sense to encode a tuple as a list rather than letting it slip to repr?
```suggestion
    elif type(obj) in (list, tuple):
```

Reply:
^ mostly trying out the new GH suggest change feature :wink:

Reply:
Makes sense, but also want that to serialize as a tuple!